### ISSUE TITLE
What is one iteration?

### ISSUE THREAD

**Issue by natasasdj** [facebookresearch/Detectron#447]
Does one iteration make a pass through all training data or only through one mini batch?
When I train my custom data, the training lasts similar if I have 1K and if I have 30K. I have the same number of iterations. So, I am confused what one iteration is.

**Comment by ppwwyyxx** [comment]
In this code base, "one iteration" will train on `config.NUM_GPUS * config.IMS_PER_BATCH` images.

**Comment by ir413** [comment]
One iteration refers to one mini-batch. Note that the concept of mini-batch in the Fast/Faster/Mask R-CNN models differs from the classification setting since different mini-batch sizes are used for different parts of the model. Namely, the network backbone uses a mini-batch size of `TRAIN.IMS_PER_BATCH` images per GPU (as @ppwwyyxx explained) while the network heads use `TRAIN.BATCH_SIZE_PER_IM` RoIs per image.

**Comment by wjbKimberly** [comment]
mark

**Comment by ninafiona** [comment]
I thought one iteration was IMS_PER_BATCH, because when I set IMS_PER_BATCH=2 and NUM_GPUS =8, the shell terminal shows an error message that 2/8 isn't divisible. @ppwwyyxx
